### ISSUE TITLE
Use a Tag or Group System for Heat Sources

### ISSUE THREAD

**Issue by Darkosto** [NovaMachina/ExNihiloSequentia#281]
Hello!

Currently, when setting up custom heat sources it's done per block. The end-result is many JEI pages. I would to propose the ability to use tags or a group system to add multiple blocks to a single heat source json file. For example, I have 20 different torch items and I'd like JEI to display this as 1 window rather than listing all 20 out. Having too many pages ends up as bad user experience. If they could be sorted and condensed in JEI, that would be much simpler. 

Basically something like this: 
```json
{
  "type": "exnihilosequentia:heat",
  "tag": "forge:torches",
  "amount": 2
}
```

or listing all of them out in the file. If this feature already exists please let me know so I can close this issue. 

Thank you!

[AB#99](https://dev.azure.com/novamachina/9b4371f9-393b-4d7d-8cb9-1f6c864e3511/_workitems/edit/99)
